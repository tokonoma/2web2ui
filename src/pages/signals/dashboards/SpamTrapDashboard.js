import React, { Component } from 'react';
import { connect } from 'react-redux';
import { list as getSubaccounts } from 'src/actions/subaccounts';
import { Grid, Panel } from 'src/components/matchbox';
import Page from '../components/SignalsPage';
import SpamTrapOverview from '../containers/SpamTrapOverviewContainer';
import FacetFilter from '../components/filters/FacetFilter';
import DateFilter from '../components/filters/DateFilter';
import SubaccountFilter from '../components/filters/SubaccountFilter';
import facets from '../constants/facets';
import _ from 'lodash';
import { SPAM_TRAP_INFO } from '../constants/info';
import { PageDescription } from 'src/components/text';

export class SpamTrapDashboard extends Component {
  componentDidMount() {
    this.props.getSubaccounts();
  }

  render() {
    const { subaccounts } = this.props;

    return (
      <Page title="Spam Trap Monitoring">
        <PageDescription>{SPAM_TRAP_INFO}</PageDescription>

        <Panel>
          <Panel.Section>
            <Grid>
              <Grid.Column xs={12} md={4}>
                <DateFilter label="Date Range" />
              </Grid.Column>
            </Grid>
          </Panel.Section>
          <Panel.Section>
            <Grid>
              <Grid.Column md={4} xs={12}>
                <SubaccountFilter label="Subaccount" />
              </Grid.Column>
              {/* eslint-disable-next-line */}
              <FacetFilter facets={_.reject(facets, facet => facet.key === 'mb_provider')} />
            </Grid>
          </Panel.Section>
        </Panel>
        <SpamTrapOverview defaults={{ perPage: 25 }} subaccounts={subaccounts} hideTitle />
      </Page>
    );
  }
}

const mapStateToProps = state => ({
  subaccounts: state.subaccounts.list,
});

const mapDispatchToProps = {
  getSubaccounts,
};

export default connect(mapStateToProps, mapDispatchToProps)(SpamTrapDashboard);
