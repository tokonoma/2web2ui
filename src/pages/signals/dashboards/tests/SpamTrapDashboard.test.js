import { shallow } from 'enzyme';
import React from 'react';
import { SpamTrapDashboard } from '../SpamTrapDashboard';

describe('Signals Spam Trap Dashboard Dashboard', () => {
  const subject = (props = {}) =>
    shallow(<SpamTrapDashboard getSubaccounts={() => {}} {...props} />);

  it('renders page', () => {
    expect(subject()).toMatchSnapshot();
  });

  it('calls getSubaccounts on mount', () => {
    const getSubaccounts = jest.fn();
    subject({ getSubaccounts });
    expect(getSubaccounts).toHaveBeenCalled();
  });
});
