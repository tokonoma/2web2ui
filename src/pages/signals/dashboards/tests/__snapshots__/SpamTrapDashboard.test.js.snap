// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signals Spam Trap Dashboard Dashboard renders page 1`] = `
<SignalsPage
  title="Spam Trap Monitoring"
>
  <PageDescription>
    
  The share over time of your total email that has been sent to known spam traps. An
  excessive rate of spam trap hits is an indicator of poor list procurement and hygiene practices.

  </PageDescription>
  <Panel>
    <Panel.Section>
      <Grid>
        <Grid.Column
          md={4}
          xs={12}
        >
          <Connect(DateFilter)
            label="Date Range"
          />
        </Grid.Column>
      </Grid>
    </Panel.Section>
    <Panel.Section>
      <Grid>
        <Grid.Column
          md={4}
          xs={12}
        >
          <Connect(Connect(SubaccountFilter))
            label="Subaccount"
          />
        </Grid.Column>
        <Connect(FacetFilter)
          facets={
            Array [
              Object {
                "key": "campaign_id",
                "label": "Campaign",
              },
              Object {
                "key": "ip_pool",
                "label": "IP Pool",
              },
              Object {
                "key": "sending_domain",
                "label": "Sending Domain",
              },
            ]
          }
        />
      </Grid>
    </Panel.Section>
  </Panel>
  <Connect(withRouter(Connect(SpamTrapOverview)))
    defaults={
      Object {
        "perPage": 25,
      }
    }
    hideTitle={true}
  />
</SignalsPage>
`;
